.github/workflows/linux.yml

```diff
@@ -0,0 +1,67 @@
+name: ManyLinux
+
+# Run CI only when a release is created, on changes to main branch, or any PR 
+# to main. Do not run CI on any other branch. Also, skip any non-source changes 
+# from running on CI
+on:
```

Comment:
I think there is an optimization to be made here. In pylint we cached a step called "prepare base" used by other jobs later, by using a key based on [the requirements, setup.py and setup.cfg](https://github.com/PyCQA/pylint/blob/master/.github/workflows/ci.yaml#L34). Disregard this comment if not applicable here. also it's probably a next step once this MR is merged.

Reply:
Good idea.